src/SprykerSdk/Zed/AppSdk/Business/Validator/AsyncApi/AsyncApiValidator.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Zed\AppSdk\Business\Validator\AsyncApi;
+
+use Behat\Gherkin\Exception\ParserException;
+use Exception;
+use Generated\Shared\Transfer\MessageTransfer;
+use Generated\Shared\Transfer\ValidateRequestTransfer;
+use Generated\Shared\Transfer\ValidateResponseTransfer;
+use SprykerSdk\Zed\AppSdk\Business\Validator\AbstractValidator;
+use Symfony\Component\Yaml\Yaml;
+
+class AsyncApiValidator extends AbstractValidator
+{
+    /**
+     * @param \Generated\Shared\Transfer\ValidateRequestTransfer $validateRequestTransfer
+     * @param \Generated\Shared\Transfer\ValidateResponseTransfer|null $validateResponseTransfer
+     *
+     * @throws \Behat\Gherkin\Exception\ParserException
+     *
+     * @return \Generated\Shared\Transfer\ValidateResponseTransfer
+     */
+    public function validate(
+        ValidateRequestTransfer $validateRequestTransfer,
+        ?ValidateResponseTransfer $validateResponseTransfer = null
+    ): ValidateResponseTransfer {
+        $validateResponseTransfer ??= new ValidateResponseTransfer();
+
+        if (!$this->finder->hasFiles($validateRequestTransfer->getAsyncApiPathOrFail())) {
+            $messageTransfer = new MessageTransfer();
+            $messageTransfer->setMessage('No asyncApi files found.');
+            $validateResponseTransfer->addError($messageTransfer);
+
+            return $validateResponseTransfer;
+        }
+
+        try {
+            Yaml::parseFile($validateRequestTransfer->getAsyncApiPathOrFail());
+            $validateResponseTransfer = $this->validateFileData(Yaml::parseFile($validateRequestTransfer->getAsyncApiPathOrFail()), 'asyncapi.yml', $validateResponseTransfer);
```

Comment:
It would be enough to parse the file once and pass the content to the `validateFileData` method. You also should not hardcode the file name. Please pass the one that's given in the request transfer.